test/run

```diff
@@ -0,0 +1,188 @@
+  run "ct_test_response '${cip}:8080' 403 '.*'"
+  run "docker logs $cid | grep -s mpm_${mpm}':notice.*resuming normal operations'"
+}
+
+function run_mpm_config_test() {
+  for m in worker event prefork; do
+    _run_mpm_config_test $m
+  done
+}
+
+function run_s2i_test() {
+  # Test s2i use case
+  # Since we built the candidate image locally, we don't want S2I attempt to pull
+  # it from Docker hub
+  run "ct_s2i_usage ${IMAGE_NAME} ${s2i_args}" 0 "Testing 's2i usage'"
+  run "ct_s2i_build_as_df file://${test_dir}/sample-test-app ${IMAGE_NAME} ${IMAGE_NAME}-testapp ${s2i_args}" 0 "Testing 's2i build'"
+  CONTAINER_ARGS='--user 1000' IMAGE_NAME=${IMAGE_NAME}-testapp ct_create_container testing-app-s2i
+  cip=$(ct_get_cip 'testing-app-s2i')
+  run "ct_test_response '${cip}:8080' 200 'This is a sample s2i application with static content.'"
+}
+
+function run_pre_init_test() {
+  # Test s2i use case #2 - testing pre-init script
+  # Since we built the candidate image locally, we don't want S2I attempt to pull
+  # it from Docker hub
+  run "ct_s2i_build_as_df file://${test_dir}/pre-init-test-app ${IMAGE_NAME} ${IMAGE_NAME}-testapp2 ${s2i_args}" 0 "Testing 's2i build' with pre-init script"
+  CONTAINER_ARGS='--user 1000' IMAGE_NAME=${IMAGE_NAME}-testapp2 ct_create_container testing-app-pre-init
+  cip=$(ct_get_cip 'testing-app-pre-init')
+  run "ct_test_response '${cip}:8080' 200 'This content was replaced by pre-init script.'"
+}
+
+function run_self_cert_test() {
+  # Test s2i use case #3 - using own ssl certs
+  # Since we built the candidate image locally, we don't want S2I attempt to pull
+  # it from Docker hub
+  run "ct_s2i_build_as_df file://${test_dir}/self-signed-ssl ${IMAGE_NAME} ${IMAGE_NAME}-self-signed ${s2i_args}" 0 "Testing 's2i build' with self-signed cert"
+  CONTAINER_ARGS='--user 1000' IMAGE_NAME=${IMAGE_NAME}-self-signed ct_create_container testing-self-signed
+  cip=$(ct_get_cip 'testing-self-signed')
+  run "ct_test_response '${cip}:8080' 200 '.*'"
+  run "curl -k https://${cip}:8443 > output_ssl_cert"
+  run "fgrep -e 'SSL test works' output_ssl_cert"
+  echo | openssl s_client -showcerts -servername ${cip} -connect ${cip}:8443 2>/dev/null | openssl x509 -inform pem -noout -text >./servercert
+  openssl x509 -in ${test_dir}/self-signed-ssl/httpd-ssl/certs/server-cert-selfsigned.pem -inform pem -noout -text >./configcert
+  run "diff ./configcert ./servercert"
+  run "diff ./configcert ./servercert >cert.diff"
+}
+
+function run_dockerfiles_test() {
+  run "ct_test_app_dockerfile test/examples/Dockerfile 'https://github.com/sclorg/httpd-ex.git' 'Welcome to your static httpd application on OpenShift' app-src" 0
+  run "ct_test_app_dockerfile test/examples/Dockerfile.s2i 'https://github.com/sclorg/httpd-ex.git' 'Welcome to your static httpd application on OpenShift' app-src" 0
+}
+
+ct_init
+
+s2i_args="--pull-policy=never"
+
+run "docker inspect $IMAGE_NAME >/dev/null || docker pull $IMAGE_NAME" 0
+
+
+TEST_LIST="\
```

Comment:
This should be at the beginning of the file, as we have it in all other containers.